### ISSUE TITLE
last shipped rc files now available!

### ISSUE THREAD

**Issue by craigwi** [Microsoft/winfile#193]
Hey folks, my contacts in Microsoft found the last known localized releases of winfile.exe in NT4 (in sp4, sp5 and sp6a).  There were 19 languages represented.  I extracted the resources from each binary and added the .rc file to the directory src\origrc.  

I have not compared those files to the translations done thus far.  Have at it!

**Comment by clzls** [comment]
I've looked into the translations of both zh_CN and zh_TW.

As for zh_CN and zh_TW translations, they are, generally, worse than current translations. In fact, I guess the original version of zh_CN translations in src\lang was exactly the one in src\origrc and with our patches the current version of zh_CN translations is perfect enough.

For two different writing systems of Chinese, Simplified Chinese(zh_CN) used in mainland China and Traditional Chinese(zh_TW) for Taiwan, Macao and HK, are we going to maintain both of them or just provide Simplified Chinese, since people using either system can simply understand another?

If you decide to add zh_TW, you can use the rc in src\origrc as a base version. That's readable enough.

**Comment by craigwi** [comment]
Hi @clzls, I will defer to those with fluency in each of the original languages to propose any changes.  Given your efforts on the zh_CH translation, I look to you for guidance.

**Comment by fuzun** [comment]
Hey @craigwi, It looks like there are more languages embedded as legacy resource format into older (16 bit) version of winfile exe. For example winfile of Windows 3.1 has more languages.

They can be easily extracted as RES files out of the executable binary then decompiled to RC but I don't know if it's allowed. Maybe you can ask your friends and they can provide the files.

**Comment by craigwi** [comment]
I found the files from a variety of Win3.1 versions.  The additional languages appear to include Catalan, Greek, Hebrew, Persian, Thai, and Turkish.  There are also some variations of French, a second Portuguese version and one called PanEuro.

A few .ex_ files didn't expand correctly and I'm not sure which tools to use to extract the RES files and decompile.  Care to point me to them?

Thanks.

**Comment by fuzun** [comment]
Yeah it seems we are talking about the same version. The Win3.1 version includes languages you listed.

I did an initial test for Turkish language and I got a correct Windows-1254 encoded rc file as result.

I have used "Borland Resource Workshop" to extract RES file and decompile it to rc (I could not find any other application that can work with dos mz executables). Note that I had to use Dosbox to run that application on a Win3.1 system.

Result (rc file) for Turkish language:
https://gist.github.com/fuzun/0d49c753eacf4469d1b79dc6279ead5d

Since I already have a set-up system, I can also provide all of the rc files. But these RC files are not compatible with current ecosystem. So this should be the last option for retrieving language files. 

I wonder maybe the problem with "ex_" files not expanding correctly is related to encoding ? Can you try Turkish with Windows-1254 encoding first ?

Also, if there is no automated process for rebuilding new languages from old rc files, I can work for Turkish language.

**Comment by craigwi** [comment]
Thanks @fuzun for your offer.  The newer 32-bit expand.exe doesn't work right on some old files.  I ended up running the original 16-bit expand.exe to get all of the decompressed files.

Also, I found the source code for an ancient sample tool (exeview) here at Microsoft which reads 16-bit Windows exes, including resources.  I need to write a bit of code to output the results in a format similar to the way VS outputs it.

**Comment by thecatkitty** [comment]
I've compared Polish translation files. It is exactly the version I've used, so there's no need to merge or fix.

**Comment by craigwi** [comment]
I finished the extraction process mentioned above.  The commit 1c577a1ce2ddc2ebaa91aa870022c0267beee7d4 contains all of the non-English versions from Win 3.1 or WfW 3.11.  I also included the US English version for reference.  The file names were derived from the directory in which the binary was located (e.g., Brazil).  The output only includes the menu, dialog and string tables.  The rest of the resources (e.g., bitmap) are mostly unchanged :).

I will add the tool I used to do the extraction soon.
